# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/wrench/reftests/blend/child-surface.yaml
# Commit: 8c70bc92ad36
# Full Hash: 8c70bc92ad36beb61f2cd6031e80ace3793fe53b
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-02-24 03:51:41
# Regressor Bug: 1749380
# File Overlap Count: 1
# Description:
#   Bug 1749380 - Part 1 - Improve how WR handles bounding rects for off-screen surfaces r=gfx-reviewers
#   
#   This patch introduces a number of subtle but important changes
#   to how we deal with off-screen surfaces. The overall goals are:
#   
# ==============================================================================

diff -r c6f1243c3ba4 -r 8c70bc92ad36 gfx/wr/wrench/reftests/blend/child-surface.yaml
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/wr/wrench/reftests/blend/child-surface.yaml	Wed Feb 23 20:49:26 2022 +0000
@@ -0,0 +1,22 @@
+# verify that the clipping_rect of a child surface (caused by the identity filter)
+# is corrected used to select the backdrop for a mix-blend child surface
+---
+root:
+  items:
+    - type: stacking-context
+      filters: [identity]
+      items:
+      - type: stacking-context
+        blend-container: true
+        bounds: [0, 0, 100, 100]
+        items:
+          - type: rect
+            bounds: [0, 0, 100, 100]
+            color: [255, 0, 0]
+          - type: stacking-context
+            bounds: [0, 0, 100, 100]
+            mix-blend-mode: hue
+            items:
+              - type: rect
+                bounds: [0, 0, 100, 100]
+                color: [255, 255, 0]